dataflow_test.py

```diff
@@ -307,6 +307,13 @@ def test_spy():
 
 
 def test_spy_count():
+
+    # count is a component that can be needed in the middle
+    # of a pipeline. However, because it is a sink it needs
+    # to be plugged into a spy. Thus, the component spy_count
+    # provides a comfortable interface to access the future
+    # and spy objects in a single line.
```

Comment:
I think that we should replace all of these by a single, pithily-named, sink-to-spy converter.

As for the wording ... well, let's talk about it when we get there.

Reply:
I guess you mean futuresink-to-futurespy? sink-to-spy would just be `df.branch`.

Reply:
Yup.

Makes keeping the name pithy a bit more difficult, though.

Note to self: sink-to-spy = df.branch needs to be an example in the documentation somewhere, if it isn't already.

Reply:
Suggestions for the name? I think `sink_to_spy` is misleading. `fsink_to_fspy` is not as obvious as I would like.

Reply:
Let me start writing the wordy documentation (which I've been on the verge of doing for the last 2 hours), and see what inspiration I get.

Have you tried to get mdbook working locally? The Rust package situation is very good indeed, so you should have no problem. It will compile it from source, so it just takes a while.

It would be good if you could contribute to that too. I suspect that making some progress over there will clear up all sorts of things in terms of what we want to do and say in this PR.

Reply:
I haven't got into that yet. Will take a look.